mir-ci/mir_ci/robot_resources/screencopy.resource

```diff
@@ -6,49 +6,84 @@ Library             ../robot_libraries/WaylandHid.py
-    ...    {"x": int((${region}[left] + ${region}[right]) / 2), "y": int((${region}[top] + ${region}[bottom]) / 2)}
+    ...    {"x": (${region}[left] + ${region}[right]) / 2, "y": (${region}[top] + ${region}[bottom]) / 2}
     RETURN    ${point}
 
-Displace
+Displace ${point} By ${displacement_x} And ${displacement_y}
     [Documentation]    Shift a point by the specified displacements along the x and y axes.
-    [Arguments]    ${point}    ${displacement_x}    ${displacement_y}
     ${point}=    Evaluate    {"x": ${point}[x] + ${displacement_x}, "y": ${point}[y] + ${displacement_y}}
     RETURN    ${point}
 
-Move Pointer To Template
-    [Documentation]    Move the pointer to the center of the first matching image template.
-    [Arguments]    ${template}
-    ${regions}=    Match    ${template}
-    ${center}=    Get Center    ${regions}[0]
-    Move Pointer To Absolute    ${center}[x]    ${center}[y]
-    RETURN    ${center}
-
-Walk Pointer From ${start_position} To Template
-    [Documentation]    Move the pointer in incremental steps from a start position
-    ...    to the center of the first matching image template.
-    ...    If ${steps} <= 0, the pointer is moved in increments of 16 pixels.
-    ...    ${step_sleep} is the time to sleep after each step, in seconds.
+Move Pointer To ${destination}
+    [Documentation]    Move the pointer to a position or image template.
+    ...
+    ...    If ${destination} is an image template, the corresponding position
+    ...    will be that of the center of the first matching template region.
+    ...
+    ...    The return value is the pointer position after the move.
+    IF    ${{isinstance($destination, str)}}
+        ${regions}=    Screencopy.Match    ${destination}
+        ${position}=    Get Center Of ${regions}[0]
+    ELSE
+        ${position}=    Set Variable    ${destination}
+    END
+    WaylandHid.Move Pointer To Absolute    ${position}[x]    ${position}[y]
+    RETURN    ${position}
+
+Move Pointer To Proportional ${position}
+    [Documentation]    Move the pointer to a position given as proportions to
+    ...    the size of the display output.
+    ...
+    ...    The return value is the pointer position after the move.
+    WaylandHid.Move Pointer To Proportional    ${position}[x]    ${position}[y]
+    RETURN    ${position}
+
+Walk Pointer From ${start_position} To ${destination}
+    [Documentation]    Move the pointer in incremental steps from a start
+    ...    position to a destination position or image template.
+    ...
+    ...    If an image template is given as ${destination}, the corresponding
+    ...    position will be that of the center of the first matching template
+    ...    region.
+    ...
+    ...    Optional arguments:
+    ...    ${steps}: number of steps. If <= 0, the pointer is moved in
+    ...    increments of 16 pixels. Default is 0.
+    ...    ${step_sleep}: time to sleep after each step, in seconds.
+    ...    Default is 0.
```

Comment:
Maybe `${delay}`?

Reply:
Sounds good. TBH when I first wrote the keyword, I thought about using `delay` but I was afraid that it might be confused with an initial delay rather than a delay after each step. I wouldn't argue against its use.